bigtable/client/row_reader.cc

```diff
@@ -129,7 +135,7 @@ void RowReader::Advance(internal::OptionalRow& row) {
       status = grpc::Status(grpc::INTERNAL, ex.what());
     }
 #else
-    status = AdvanceOrFail(row);
+    status_ = status = AdvanceOrFail(row);
 #endif  // GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
 
     if (status.ok()) {
```

Comment:
Further down (line 159) the code raises exceptions.  Are you going to submit a separate PR for #247 or is this PR supposed to solve that too?  A future change is Okay.

Reply:
I missed that exception, Fixed it now. Included row reader change as well in this PR to solve  #247 Included finish function in iterator as per no exception guideline. Let me know if you want me to separate PR for that.